### PR TITLE
add e2e test for dualstack clusters on azure

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What does this PR do / Why do we need it**:
Adds e2e test to verify dualstack setup of Azure for different OSs.
Makes it easy to add other cloud providers to the test. 

**Does this PR close any issues?**:
Part of https://github.com/kubermatic/kubermatic/issues/9617

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: addons/cilium/cilium.yaml [hunk 1]
```diff
@@ -143,11 +143,10 @@ data:
   sidecar-istio-proxy-image: "cilium/istio_proxy"
 
   # Name of the cluster. Only relevant when building a mesh of clusters.
-  cluster-name: default
+  cluster-name: defaultisbad
```
Comment (PratikDeoghare) [inline addons/cilium/cilium.yaml]: dd

File: pkg/addon/template.go [hunk 2]
```diff
@@ -218,9 +222,15 @@ type ClusterNetwork struct {
...
+	ClusterName string
+}
+
 type CNIPlugin struct {
-	Type    string
-	Version string
+	Type          string
+	Version       string
+	CiliumOptions *CiliumOptions
 }
```
Comment (PratikDeoghare) [inline pkg/addon/template.go]: dd

File: pkg/test/dualstack/azure_dualstack_test.go [hunk 3]
```diff
@@ -0,0 +1,508 @@
+package dualstack
```
Comment (PratikDeoghare) [inline pkg/test/dualstack/azure_dualstack_test.go:19]: add licence and build  tag

File: .prow/features.yaml [hunk 4]
```diff
@@ -125,6 +125,43 @@ presubmits:
             limits:
               memory: 6Gi
 
+  - name: pre-kubermatic-dualstack-e2e
+    always_run: true # TODO: remove before merging
```
Comment (rastislavs) [inline .prow/features.yaml]: remove?

File: hack/ci/testdata/kubermatic_dualstack.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  name: e2e
+  namespace: kubermatic
+spec:
+  ingress:
+    domain: '__KUBERMATIC_DOMAIN__'
+    disable: true
+  imagePullSecret: '__IMAGE_PULL_SECRET__'
+  userCluster:
+    apiserverReplicas: 2
+    addons:
```
Comment (rastislavs) [inline hack/ci/testdata/kubermatic_dualstack.yaml]: do we need this addon list?

Comment (PratikDeoghare) [reply hack/ci/testdata/kubermatic_dualstack.yaml]: Probably not. This was copied from the cilium test.

File: .prow/features.yaml (hunk 4 above)
Comment (PratikDeoghare) [reply .prow/features.yaml]: Yes yes. Will replace with appropriate regex.

File: pkg/test/dualstack/azure_dualstack_test.go [hunk 6]
```diff
@@ -0,0 +1,581 @@
...
+				},
+			},
+			CniPlugin: &models.CNIPluginSettings{
+				Version: "v1.11",
+				Type:    "cilium",
+			},
+			ClusterNetwork: &models.ClusterNetworkingConfig{
+				NodeCIDRMaskSizeIPV4: 24,
+				// {"cidrBlocks":["172.25.0.0/16","fd00::/104"]},
+				// "services":{"cidrBlocks":["10.240.16.0/20","fd03::/120"]}
```
Comment (rastislavs) [inline pkg/test/dualstack/azure_dualstack_test.go]: the comment does not seem to have any special meaning, I guess you could remove it

**Review by ahmedwaleedmalik — COMMENTED** [review]
Looks super good but can you please rebase your branch? Currently, there are a lot of ineffective commit messages like `.` etc. They get squashed but the messages are aggregated in the commit message.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve 🏎️

**Pull request merged**